src/sandstorm/web-session.capnp

```diff
@@ -97,6 +107,32 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
 
     accept @2 :List(AcceptedType);
     # This corresponds to the Accept header
+
+    additionalHeaders @3 :List(Header);
+    # Additional headers present in the request. Only whitelisted headers are
+    # permitted.
+
+    struct Header {
+      name @0 :Text;  # lower-cased name
+      value @1 :Text;
+    }
+
+    const headerWhitelist :List(Text) = [
+      "if-match",              # webdav etag
+      "if-not-match",          # webdav etag
+      "depth",                 # webdav PROPFIND
+      "destination",           # webdav COPY/MOVE
```

Comment:
Hmm, I'm concerned that this header is specified as containing an absolute URL to which the resource is to be moved. This interacts weirdly with Sandstorm hosts being generally unspecified.

I wonder if we want to explicitly encode this header in the parameters to copy() and move(), and specify that it should be a locally-absolute path but not specify a scheme nor hostname?